cupy/linalg/_decomposition.py

```diff
@@ -352,6 +354,63 @@ def qr(a, mode='reduced'):
     return q, _util._triu(r)
 
 
+def _svd_batched(a, a_dtype, full_matrices, compute_uv):
+    batch_shape = a.shape[:-2]
+    batch_size = internal.prod(batch_shape)
+    n, m = a.shape[-2:]
+    s_dtype = a_dtype.lower()
+
+    # first handle any 0-size inputs
+    if batch_size == 0:
+        k = min(m, n)
+        s = cupy.empty(batch_shape + (k,), s_dtype)
+        if compute_uv:
+            if full_matrices:
+                u = cupy.empty(batch_shape + (n, n), dtype=a_dtype)
+                vt = cupy.empty(batch_shape + (m, m), dtype=a_dtype)
+            else:
+                u = cupy.empty(batch_shape + (n, k), dtype=a_dtype)
+                vt = cupy.empty(batch_shape + (k, m), dtype=a_dtype)
+            return u, s, vt
+        else:
+            return s
+    elif m == 0 or n == 0:
+        s = cupy.empty(batch_shape + (0,), s_dtype)
+        if compute_uv:
+            if full_matrices:
+                u = [cupy.eye(n, dtype=a_dtype) for _ in range(batch_size)]
+                u = cupy.stack(u, axis=0).reshape(*batch_shape, n, n)
+                vt = [cupy.eye(m, dtype=a_dtype) for _ in range(batch_size)]
+                vt = cupy.stack(vt, axis=0).reshape(*batch_shape, m, m)
+            else:
+                u = cupy.empty(batch_shape + (n, 0), dtype=a_dtype)
+                vt = cupy.empty(batch_shape + (0, m), dtype=a_dtype)
+            return u, s, vt
+        else:
+            return s
+
+    # ...then delegate real computation to cuSOLVER
+    a = a.reshape(-1, *(a.shape[-2:]))
+    if runtime.is_hip or (m <= 32 and n <= 32):
+        # copy is done in _gesvdj_batched, so let's try not to do it here
+        a = a.astype(a_dtype, order='C', copy=False)
+        out = _gesvdj_batched(a, full_matrices, compute_uv, False)
+    else:
+        # manually loop over cusolverDn<t>gesvd()
+        # copy (via possible type casting) is done in _gesvd_batched
+        out = _gesvd_batched(a, a_dtype, full_matrices, compute_uv, False)
+    if compute_uv:
+        u, s, v = out
```

Comment:
Good catch, @toslunar! There was actually a bug related to this, which I fixed in 62e11ac. Yes, it returns `V` instead of `V^H` (transpose conjugate).